src/sensor.cpp

```diff
@@ -267,11 +267,12 @@ void log_callback_end( uint32_t fps,
     {
         auto system_time = environment::get_instance().get_time_service()->get_time();
         auto fr = std::make_shared<frame>();
-        byte* pix = (byte*)fo.pixels;
-        std::vector<byte> pixels(pix, pix + fo.frame_size);
-        fr->data = pixels;
+        
+        //REMOVED! - no need to add 2 copies to a frame
+        //byte* pix = (byte*)fo.pixels;
+        //std::vector<byte> pixels(pix, pix + fo.frame_size);
+        //fr->data = pixels;
```

Comment:
This is dangerous.
This leaves `fr->data` (for regular frames) basically uninitialized (empty), right? I.e., as long as it's not used, we're OK. So why not remove it?
I believe a composite frame still uses it... maybe we can move it there?
(not for this PR, but general thought)

Reply:
I agree, like we said the `generate_frame_from_data()` needs to be refactored